modules/core/src/main/scala/fs2/kafka/KafkaProducer.scala

```diff
@@ -132,26 +132,21 @@ object KafkaProducer {
     valueSerializer: Serializer[F, V],
     producer: KafkaByteProducer
   )(
-    implicit F: Async[F],
-    dispatcher: Dispatcher[F]
+    implicit F: Async[F]
   ): ProducerRecord[K, V] => F[F[(ProducerRecord[K, V], RecordMetadata)]] =
     record =>
       asJavaRecord(keySerializer, valueSerializer, record).flatMap { javaRecord =>
-        Deferred[F, Either[Throwable, (ProducerRecord[K, V], RecordMetadata)]].flatMap { deferred =>
+        F.delay(Promise[(ProducerRecord[K, V], RecordMetadata)]()).flatMap { promise =>
           F.blocking {
```

Comment:
As far as I understand, in ce3 code in this block will execute on the default blocking thread-pool. But it should work on a dedicated thread-pool, not default.

Reply:
This is Producer, so the default blocking pool is fine. But this does make me notice a problem with #590 - if the user provided a custom pool then we should be using it.

Reply:
Sounds good. We should fix this before 2.0 final release.